src/libs/ajax/User.ts

```diff
@@ -0,0 +1,168 @@
+import * as fp from 'lodash/fp';
+import { cloneDeep, flow, unset } from 'lodash/fp';
+import { Config } from '../config';
+import axios from 'axios';
+import { getApiUrl, fetchOk, fetchAny } from '../ajax';
+import { CreateDuosUserResponse, DuosUserResponse, UpdateDuosUserResponse } from 'src/types/responseTypes';
+import { CreateDuosUserRequest, UpdateDuosUserRequestV1, UpdateDuosUserRequestV2 } from 'src/types/requestTypes';
+import { AcknowledgementMap, ApprovedDataset, Dataset, DuosUser, SimplifiedDuosUser } from 'src/types/model';
+
+export const User = {
+  getMe: async (): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/me`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  getById: async (id: number): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/${id}`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  list: async (roleName: 'Admin' | 'SigningOfficial'): Promise<DuosUserResponse[]> => {
+    const url = `${await getApiUrl()}/api/user/role/${roleName}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return res.json();
+  },
+
+  create: async (request: CreateDuosUserRequest): Promise<CreateDuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/dacuser`;
+    try {
+      const res = await fetchOk(
+        url,
+        fp.mergeAll([
+          Config.authOpts(),
+          Config.jsonBody(request),
+          { method: 'POST' },
+        ])
+      );
+      if (res.ok) {
+        return res.json();
+      }
+    } catch (err) {
+      return false;
+    }
+  },
+
+  updateSelf: async (payload: UpdateDuosUserRequestV1): Promise<UpdateDuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user`;
+    // We should not be updating the user's create date, associated institution, or library cards
+    try {
+      const res = await fetchOk(
+        url,
+        fp.mergeAll([
+          Config.authOpts(),
+          Config.jsonBody(payload),
+          { method: 'PUT' },
+        ])
+      );
+      if (res.ok) {
+        return res.json();
+      }
+    } catch (err) {
+      return false;
+    }
+  },
+
+  update: async (user: UpdateDuosUserRequestV2, userId: number)/*: Promise<UpdateDuosUserResponse>*/ => {
+    const url = `${await getApiUrl()}/api/user/${userId}`;
+    // We should not be updating the user's create date, associated institution, or library cards
+    // This below code does not seem to work at all and
+    // does not seem appropriate for this request anyway.
+    // The UpdateDuosUserRequestV2 is not the same shape as a DuosUser
+    // like this flow suggests.
+    const filteredUser = flow(
+      cloneDeep,
+      unset('updatedUser.createDate'),
+      unset('updatedUser.institution'),
+      unset('updatedUser.libraryCards')
+    )(user);
```

Comment:
This seems to be a relic of an older implementation of this API. I don't think it is within the scope of this ticket to update/remove this logic, however, I did want to check with the team about this before spinning it off into its own ticket

Reply:
To be more specific (and I can add this to the comment directly if desired), these unset calls do not in fact unset those properties but the payload being used in the update request does not have those fields anyway.

Reply:
This is an artifact of the older API. The latest version accepts a limited number of fields that can be updated as per the API docs. I think it is appropriate to leave this logic as is and we can address it in a future ticket.
* [UserUpdateFields](https://github.com/DataBiosphere/consent/blob/8305d4748f6c6589fc68723de759c871c9a2283e/src/main/java/org/broadinstitute/consent/http/models/UserUpdateFields.java#L15)
* [Current API Docs](https://github.com/DataBiosphere/consent/blob/8305d4748f6c6589fc68723de759c871c9a2283e/src/main/resources/assets/paths/userByIdv2.yaml#L4)